mlir/include/mlir/Dialect/LLVMIR/GENXOps.td

```diff
@@ -148,14 +148,71 @@ def GENX_BarrierOp : GENX_Op<"barrier"> {
   let assemblyFormat = "attr-dict";
 }
 
-def ShflKindXor : I32EnumAttrCase<"XOR", 0>;
-def ShflKindUp   : I32EnumAttrCase<"UP", 1>;
-def ShflKindDown : I32EnumAttrCase<"DOWN", 2>;
-def ShflKindIdx  : I32EnumAttrCase<"IDX", 3>;
+/// Enum attribute of the different memory fence flags.
+def MemoryFenceFlag : I32EnumAttr<"MemoryFenceFlag", "GENX memory fence flag",
+  [
+    I32EnumAttrCase<"CLK_LOCAL_MEM_FENCE",  1>,
+    I32EnumAttrCase<"CLK_GLOBAL_MEM_FENCE", 2>,
+    I32EnumAttrCase<"CLK_IMAGE_MEM_FENCE",  4>
+  ]> {
+  let genSpecializedAttr = 0;
+  let cppNamespace = "::mlir::GENX";
+}
+def MemoryFenceFlagAttr : EnumAttr<GENX_Dialect, MemoryFenceFlag, "memory_fence_flag">;
+
+/// Enum attribute of the different memory order.
+def MemoryOrder : I32EnumAttr<"MemoryOrder", "GENX memory order",
+  [
+    I32EnumAttrCase<"Relaxed",                 0>,
+    I32EnumAttrCase<"Acquire",                 2>,
+    I32EnumAttrCase<"Release",                 3>,
+    I32EnumAttrCase<"AcquireRelease",          4>,
+    I32EnumAttrCase<"SequentiallyConsistent",  5>
+  ]> {
+  let genSpecializedAttr = 0;
+  let cppNamespace = "::mlir::GENX";
+}
+def MemoryOrderAttr : EnumAttr<GENX_Dialect, MemoryOrder, "memory_order">;
+
+/// Enum attribute of the different memory scope.
+def MemoryScope : I32EnumAttr<"MemoryScope", "GENX memory scope",
+  [
+    I32EnumAttrCase<"work_item",        0>,
+    I32EnumAttrCase<"work_group",       1>,
+    I32EnumAttrCase<"device",           2>,
+    I32EnumAttrCase<"all_svm_devices",  3>,
+    I32EnumAttrCase<"sub_group",        4>
+  ]> {
+  let genSpecializedAttr = 0;
+  let cppNamespace = "::mlir::GENX";
+}
+def MemoryScopeAttr : EnumAttr<GENX_Dialect, MemoryScope, "memory_scope">;
+
+def GENX_AtomicWorkItemFenceOp : GENX_Op<"atomic_work_item_fence">,
+  Arguments<(ins I32Attr:$flags,
+                 MemoryOrderAttr:$order,
+                 MemoryScopeAttr:$scope)> {
+  let summary = "Atomic work item fence";
+  string baseDescription = [{
+    Ensures that memory accesses issued before this instruction will be observed
```

Comment:
Need to expand documentation or users will not know what flags, order, etc... are.

Reply:
done.